jetty-core/jetty-siwe/src/main/java/org/eclipse/jetty/security/siwe/EthereumAuthenticator.java

```diff
@@ -0,0 +1,772 @@
+
+    private String _loginPath;
+    private String _authPath = DEFAULT_AUTH_PATH;
+    private String _noncePath = DEFAULT_NONCE_PATH;
+    private long _maxMessageSize = 4 * 1024;
+    private String _logoutRedirectPath;
+    private String _errorPage;
+    private String _errorPath;
+    private String _errorQuery;
+    private boolean _dispatch;
+
+    public EthereumAuthenticator()
+    {
+    }
+
+    public void includeDomains(String... domains)
+    {
+        _domains.include(domains);
+    }
+
+    public void includeSchemes(String... schemes)
+    {
+        _schemes.include(schemes);
+    }
+
+    public void includeChainIds(String... chainIds)
+    {
+        _chainIds.include(chainIds);
+    }
+
+    @Override
+    public void setConfiguration(Authenticator.Configuration authConfig)
+    {
+        String loginPath = authConfig.getParameter(LOGIN_PATH_PARAM);
+        if (loginPath != null)
+            setLoginPath(loginPath);
+
+        String authPath = authConfig.getParameter(AUTH_PATH_PARAM);
+        if (authPath != null)
+            setAuthPath(authPath);
+
+        String noncePath = authConfig.getParameter(NONCE_PATH_PARAM);
+        if (noncePath != null)
+            setNoncePath(noncePath);
+
+        String maxMessageSize = authConfig.getParameter(MAX_MESSAGE_SIZE_PARAM);
+        if (maxMessageSize != null)
+            setMaxMessageSize(Integer.parseInt(maxMessageSize));
+
+        String logout = authConfig.getParameter(LOGOUT_REDIRECT_PARAM);
+        if (logout != null)
+            setLogoutRedirectPath(logout);
+
+        String error = authConfig.getParameter(ERROR_PAGE);
+        if (error != null)
+            setErrorPage(error);
+
+        String dispatch = authConfig.getParameter(DISPATCH_PARAM);
+        if (dispatch != null)
+            setDispatch(Boolean.parseBoolean(dispatch));
+
+        // If no LoginService is set we allow any user to log in.
+        if (authConfig.getLoginService() == null)
+        {
+            LoginService loginService = new AnyUserLoginService();
+            authConfig = new Configuration.Wrapper(authConfig)
+            {
+                @Override
+                public LoginService getLoginService()
+                {
+                    return loginService;
+                }
+            };
+        }
+
+        super.setConfiguration(authConfig);
+    }
+
+    @Override
+    public String getAuthenticationType()
+    {
+        return Authenticator.SIWE_AUTH;
+    }
+
+    public void setLoginPath(String loginPath)
+    {
+        if (loginPath == null)
+        {
+            LOG.warn("login path must not be null, defaulting to " + _loginPath);
+            loginPath = _loginPath;
+        }
+        else if (!loginPath.startsWith("/"))
+        {
+            LOG.warn("login path must start with /");
+            loginPath = "/" + loginPath;
+        }
+
+        _loginPath = loginPath;
+    }
+
+    public void setAuthPath(String authPath)
```

Comment:
I prefer long form of `Auth` to distinguish between authentication and authorization.

Reply:
renamed everywhere to `authenticationPath`.